### ISSUE TITLE
Volar-ized Glint Roadmap

### ISSUE THREAD

**Issue by machty** [typed-ember/glint#728]
## Overview

This roadmap absorbs the contents of the now-closed [Language Tooling RFC](https://github.com/machty/rfcs/blob/1c984ab3685d8e539b279163c7a9ac41b92e0a79/text/0000-ember-language-tooling-roadmap.md). 

Now that the initial [Volar Refactor PR](https://github.com/typed-ember/glint/pull/726) has been merged, there a number of next steps to pursue before Glint 2.0 is ready for primetime.

## Roadmap

- [x] Fix test suite, all green tests, via test.skip() (and add to this issue for any remaining feature TODOs)
- [ ] - [x] Fix `glint --build --clean` specs (broken since Volar -- cleans everything but .gts file outputs). Johnson thinks it can be supported, needs upstream fix.
- [x] Support for .gjs (and passing tests)
- [ ] Support for .hbs and companion files
- Reinstate disabled CI:
  - [ ] eslint
  - [ ] windows (blocked on volar 2.4.0 breaking stuff, discussion [here](https://discord.com/channels/1192759067815464990/1192759067815464993/1258395162497777664))
  - [ ] type-tests
  - [ ] test-floating-deps
  - [ ] test-ts-nightly
- [ ] Figure out how to resolve conflicts/overlapping with `vsc-ember-syntax` https://github.com/lifeart/vsc-ember-syntax/issues/72
  - I need to review how/why I inlined portions of that addon into this branch; I believe it's because otherwise Glint/Volar can't determine a language ID for .gts files
- [ ] get all `{{ @glint-expect-error }}` and related directives working

- [ ] constrain `data` objects on nested embedded codes https://discord.com/channels/1192759067815464990/1252021970443440128/1252298644695158874

**Comment by machty** [comment]
CI is green; now to unskip and fix a number of specs where the underlying behavior is actually correct, but the test is broken.

**Comment by machty** [comment]
Closing this in favor of a soon to be published GH project. All of the items contained within have been extracted/converted into GH issues.

**Comment by bartocc** [comment]
@machty has the GH project been published? and if so, could you provide its URL plz?
